components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/ActionManagementServiceImpl.java

```diff
@@ -268,4 +355,83 @@ private Action updateEndpointAuthenticationProperties(String actionType, String
         return CACHE_BACKED_DAO.updateActionEndpointAuthProperties(actionType, actionId, authentication,
                 IdentityTenantUtil.getTenantId(tenantDomain));
     }
+
+    /**
+     * Perform pre validations on action model when creating an action.
+     *
+     * @param actionType Action type.
+     * @param action     Action create model.
+     * @throws ActionMgtException if action model is invalid.
+     */
+    private void doPreAddActionValidations(String actionType, Action action) throws ActionMgtException {
+
+        if (actionType.equals(Action.ActionTypes.PRE_ISSUE_ACCESS_TOKEN.getActionType())) {
+            ActionManagementUtil.isFieldEmpty(action.getName());
+            ActionManagementUtil.isFieldEmpty(action.getEndpoint().getUri());
+            ActionManagementUtil.isValidActionName(action.getName());
+            ActionManagementUtil.isValidEndpointUri(action.getEndpoint().getUri());
+            // Validate endpoint authentication details in the request.
+            Authentication endpointAuthentication = action.getEndpoint().getAuthentication();
+            String endpointAuthenticationType = endpointAuthentication.getType().getName();
+            doEndpointAuthenticationValidation(endpointAuthenticationType, endpointAuthentication);
+
+        }
+    }
+
+    /**
+     * Perform pre validations on action model when updating an existing action.
+     * This is specifically used during HTTP PATCH operation and
+     * only validate non-null and non-empty fields.
+     *
+     * @param actionType Action type.
+     * @param action Action update model.
+     * @throws ActionMgtClientException if action model is invalid.
+     */
+    private void doPreUpdateActionValidations(String actionType, Action action) throws ActionMgtClientException {
+
+        if (actionType.equals(Action.ActionTypes.PRE_ISSUE_ACCESS_TOKEN.getActionType())) {
```

Comment:
Is there any reason why we are not checking whether action properties here empty or not as in doPreAddActionValidations

Reply:
This method is specifically used for PATCH operations, where it's possible that certain fields may be left empty or null.

cc: @malithie @ashanthamara